docs/user-guide/configuring.md

```diff
@@ -382,6 +382,16 @@ The second way to use configuration files is via `.eslintrc` and `package.json`
 
 In each case, the settings in the configuration file override default settings.
 
+## Configuration File Formats
+
+ESLint supports configuration files in several formats:
+
+* **JavaScript** - use `.eslintrc.js` and export an object containing your configuration.
+* **YAML** - use `.eslintrc.yaml` or `.eslintrc.yml` to define the configuration structure.
+* **JSON** - use `.eslintrc.json` to define the configuration structure. ESLint's JSON files also allow JavaScript-style comments.
+* **package.json** - create an `eslintConfig` property in your `package.json` file and define your configuration there.
+* **Deprecated** - use `.eslintrc`, which can be either JSON or YAML.
```

Comment:
Why do we want to deprecate this?
This breaking change   (whenever it happens in future) is guaranteed to break all the projects which use eslint. As we get more and more popular/dependent upon we should avoid such breaking changes. #JustAThought